pages/_lang/generate.vue

```diff
@@ -451,6 +471,54 @@ export default class extends Vue {
     this.manifest$ = { ...this.manifest };
   }
 
+  public onClickDownloadAll() {
+
+    // local azure function
+    this.zipRequested = true;
+    const images: generator.Icon[] = [];
+    const length = this.icons.length;
+    // prevent the passing of the observer objects in the body.
+    for (let i = 0; i < length; i++) {
+      images.push({ ...this.icons[i] })
+    }
```

Comment:
This was a bit tricky and took me a while to figure out what was happening. The code was passing back an observer object with the request and not the actual list itself, hence the loop to create the list in memory to actually pass to the service.